src/btree/bt_delete.c

```diff
@@ -391,12 +391,9 @@ __wt_delete_page_skip(WT_SESSION_IMPL *session, WT_REF *ref, bool visible_all)
      * discard the page_del info either, as doing so leads to dropping the on-disk page and if the
      * prepared transaction rolls back we'd then be in trouble.
      */
-    if (visible_all)
-        skip = discard = __wt_page_del_visible_all(session, ref->page_del, true);
-    else {
-        skip = __wt_page_del_visible(session, ref->page_del, true);
-        discard = skip ? __wt_page_del_visible_all(session, ref->page_del, true) : false;
-    }
+    skip = discard = __wt_page_del_visible(session, ref->page_del, true, true);
```

Comment:
why we need to change the logic here?

Reply:
There should be no logic change, just used the new page_del_visible function.